### ISSUE TITLE
RPC ```eth_getLogs``` has ```logIndex``` as the same value for all events

### ISSUE THREAD

**Issue by leostelon** [Shardeum/shardeum-bug-reporting#10]
##### Overview

All transactions logs have a default logIndex value ```0x1``` for all events.

##### Context 

When multiple events are emitted from a 
single function call from a contract, logs produced have the same logIndex value ```0x1```.
       
Steps to Reproduce: 

1. Create a contract with multiple events and emit all those events from the same function.
2. Events emitted produce logs in the transaction which can be viewed as JSON in the explorer.
3. Every log can be seen having the same logIndex.

JSON View and check logIndex for every log example:

https://explorer-liberty20.shardeum.org/tx/0xa2d15bd942f69e6a9cad2e5469f424955dc384576d729f48d51f9c12363667ad

##### Resources

RPC method eth_getLogs specification:

https://www.quicknode.com/docs/ethereum/eth_getLogs

**Comment by MarcusWentz** [comment]
Working on Liberty 1.6. 
However, this is not working on Liberty 2.1. 

We will look into this.

**Comment by MarcusWentz** [comment]
Is this issue on Betanet 1.X? 

If so, please:

1. Share a simple contract with events you are testing with.

2. Show the logs to review the logIndex values.

**Comment by leostelon** [comment]
Simple Contract:

Test.sol
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.7;

import "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";

contract Test is ERC721URIStorage {
    constructor() ERC721("Test NFT" , "TSN") {}
    uint256 internal _assetCount = 0;

    function mintNfts() public {
        for(uint8 i=0; i < 5; i++) { // To produce 5 ERC721 Transfer events in same transaction
            _safeMint(msg.sender, _assetCount);
            _assetCount++;
        }
    }
}

Contract deployed for testing::
Contract Address: 0xC3b76De47F7638626f1A2055C0Aba52a6411Cd05
Trx Hash: 0x2097fff73e08d32935a61ef0280c529ec53557fea351ac7dc45c14eaf3c53879
Check logs and their logIndex values by clicking JSON View

**Comment by MarcusWentz** [comment]
Testing simple contract with multiple events:

```solidity
// SPDX-License-Identifier: MIT
pragma solidity 0.8.7;

contract eventsIndexTest {

    event one();
    event two();
    event three();
    event four();

    function events() public {
        emit one();
        emit two();
        emit three();
        emit four();
    }

}
```

:green_circle: Goerli: 

Contract address:

https://goerli.etherscan.io/address/0x161aeb4245695855a0e3ef43a437639d08118a1d#code

Test eth_getLogs test:

```javascript
const ethers = require("ethers");

const provider = new ethers.providers.JsonRpcProvider(process.env.goerliHTTPS_InfuraAPIKey)
const contractAddress = "0x161aeb4245695855a0e3ef43a437639d08118a1d"
const contractDeployedBlock = 8514256
const contractFirstTransactionBlock = 8514265

getLogsWithEventIndex();

async function getLogsWithEventIndex() {

  const addressTransactionLogs = await provider.getLogs({
   address: contractAddress,
   fromBlock:contractDeployedBlock,
   toBlock:contractFirstTransactionBlock,
  });

  console.log("event one() emitted at logIndex: " + addressTransactionLogs[0].logIndex);
  console.log("event two() emitted at logIndex: " + addressTransactionLogs[1].logIndex);
  console.log("event three() emitted at logIndex: " + addressTransactionLogs[2].logIndex);
  console.log("event four() emitted at logIndex: " + addressTransactionLogs[3].logIndex);

}
```

Retruns:

```shell
event one() emitted at logIndex: 100
event two() emitted at logIndex: 101
event three() emitted at logIndex: 102
event four() emitted at logIndex: 103
```

:red_circle: Betanet 1.1:

Contract address:

https://explorer-sphinx.shardeum.org/account/0xD06245458e3479aDF4bAA9d390Cf7a335226060B

Contract transaction with event logs:

https://explorer-sphinx.shardeum.org/transaction/0xc36b485971a68b1371126d1d5f93002bd40f8fd29c861a8cc59709d20f162b78

Raw JSON URL for transaction:

https://explorer-sphinx.shardeum.org/api/transaction?txHash=0xc36b485971a68b1371126d1d5f93002bd40f8fd29c861a8cc59709d20f162b78

Filter for event logs with logIndex by scraping JSON data:

```javascript
const axios = require('axios');

let baseUrl = "https://explorer-sphinx.shardeum.org/api/transaction?txHash=0xc36b485971a68b1371126d1d5f93002bd40f8fd29c861a8cc59709d20f162b78"

test_transaction_event_logIndex(baseUrl)

async function test_transaction_event_logIndex(baseUrl) {

	let responseRawJSON = await axios.get(baseUrl);
	let responseDataJSON = responseRawJSON.data;

	transactionDataReadableReceiptJSON = responseDataJSON.transactions[0].wrappedEVMAccount.readableReceipt
	console.log("event one() emitted at logIndex:", + transactionDataReadableReceiptJSON.logs[0].logIndex );
	console.log("event two() emitted at logIndex:", + transactionDataReadableReceiptJSON.logs[1].logIndex );
	console.log("event three() emitted at logIndex:", + transactionDataReadableReceiptJSON.logs[2].logIndex );
	console.log("event four() emitted at logIndex:", + transactionDataReadableReceiptJSON.logs[3].logIndex );

}
```

Returns:

```shell
event one() emitted at logIndex: 1
event two() emitted at logIndex: 1
event three() emitted at logIndex: 1
event four() emitted at logIndex: 1
```

**Comment by swarna1101** [comment]
**Issue Resolution Summary:**
- The reported bug regarding the default logIndex value has been fixed in the new network release Sphinx Dapp 1.7.1.
- Tested the scenario, and logs now have correct logIndex values.
